core/src/main/scala/com/salesforce/op/stages/impl/preparators/MinVarianceFilter.scala

```diff
@@ -0,0 +1,254 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.stages.impl.preparators
+
+import com.salesforce.op.UID
+import com.salesforce.op.features.types.{OPVector, VectorConversions}
+import com.salesforce.op.stages.base.unary.{UnaryEstimator, UnaryModel}
+import com.salesforce.op.utils.spark.RichMetadata._
+import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
+import org.apache.log4j.{Level, LogManager}
+import org.apache.spark.ml.linalg.{DenseVector, SparseVector, Vectors => NewVectors}
+import org.apache.spark.ml.param.{BooleanParam, DoubleParam, Param, Params}
+import org.apache.spark.mllib.linalg.{Vector => OldVector, Vectors => OldVectors}
+import org.apache.spark.mllib.stat.{MultivariateStatisticalSummary, Statistics}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Dataset
+import org.slf4j.impl.Log4jLoggerAdapter
+
+import scala.util.Try
+
+
+trait MinVarianceFilterParams extends Params {
+
+  final val logLevel = new Param[String](
+    parent = this, name = "logLevel",
+    doc = "sets log level (INFO, WARN, ERROR, DEBUG etc.)",
+    isValid = (s: String) => Try(Level.toLevel(s)).isSuccess
+  )
+
+  private[op] def setLogLevel(level: Level): this.type = set(logLevel, level.toString)
+
+  final val removeBadFeatures = new BooleanParam(
+    parent = this, name = "removeBadFeatures",
+    doc = "If set to true, this will automatically remove all the bad features from the feature vector"
+  )
+
+  def setRemoveBadFeatures(value: Boolean): this.type = set(removeBadFeatures, value)
+
+  def getRemoveBadFeatures: Boolean = $(removeBadFeatures)
+
+  final val minVariance = new DoubleParam(
+    parent = this, name = "minVariance",
+    doc = "Minimum amount of variance allowed for each feature"
+  )
+
+  def setMinVariance(value: Double): this.type = set(minVariance, value)
+
+  def getMinVariance: Double = $(minVariance)
+
+  setDefault(
+    removeBadFeatures -> MinVarianceFilter.RemoveBadFeatures,
+    minVariance -> MinVarianceFilter.MinVariance
+  )
+}
+
+/**
+ * The MinVarianceFilter checks that computed features have a minimum variance
+ *
+ * Like SanityChecker, the Estimator step outputs statistics on incoming data, as well as the
+ * names of features which should be dropped from the feature vector. And the transformer step
+ * applies the action of actually removing the low variance features from the feature vector
+ *
+ * Two distinctions from SanityChecker:
+ * (1) no label column as input; and
+ * (2) only filters features by variance
+ */
+class MinVarianceFilter
+(
+  operationName: String = classOf[MinVarianceFilter].getSimpleName,
+  uid: String = UID[MinVarianceFilter]
+) extends UnaryEstimator[OPVector, OPVector](operationName = operationName, uid = uid)
+  with MinVarianceFilterParams {
+
+  private def makeColumnStatistics
```

Comment:
rather than copy pasting these functions is it possible to move them to a shared companion object and just have default empty values for the parts you dont use?